### PR TITLE
dagcoin implementation

### PR DESCRIPTION
**Pull request by shubhrapahwa**
Implementation of DagCoin for BATM.

### CODE REVIEW

File: server_extensions_extra/build.gradle [hunk 1]
```diff
@@ -60,6 +60,12 @@ dependencies {
     compile(group: 'io.reactivex', name: 'rxjava', version: '1.2.4')
     compile(group: 'com.vdurmont', name: 'emoji-java', version: '3.1.3') //for chat emojis
 
+	// For DagCoin
+    compile files('/Users/shubhrapahwa/Desktop/dagCoin/atm-server/target/dagcoin-rest-client-0.0.1.jar')
```
Comment (generalbytes) [inline server_extensions_extra/build.gradle]: ???

Comment (generalbytes) [reply server_extensions_extra/build.gradle]: Client code needs to be on maven central or somewhere else.

**Comment by mikklaos** [comment]
What changes can we make so you can merge the pull request?
Pure jar file is our own code, its only in jar format so that it is not that easily readable since this repository is public. 
When will the extension improvement be ready?

**Comment by generalbytes** [comment]
Hi Mikklaos,
Your purejar (binary jar file) will never make it into this public repository. This public repository contains only opensource implementation. If your extension depends on proprietary code then keep it for your operation but please don't try to get it merged into this repo.

**Comment by mikklaos** [comment]
Do you mean that I should create purejar file and put it into some maven repo? And when code is built it is pulled in as a dependency?

Can you say where other coins are keeping their jars?

**Comment by generalbytes** [comment]
Your project has to be opensource if you want it to be part of this repo. All other coins are also opensource.

**Comment by shubhrapahwa** [comment]
Hi, we need the following jars for our implementation -
compile group: 'org.glassfish.jersey.core', name: 'jersey-client', version: '2.29'
compile group: 'org.glassfish.jersey.media', name: 'jersey-media-json-jackson', version: '2.29'
compile group: 'org.glassfish.jersey.inject', name: 'jersey-hk2', version: '2.29'

I don't see any conflicting versions of these dependencies in the project.
Please suggest if the version needs to be updated so that this code can be merged in this repo?

Also we will host the pure jar in central maven repository and update the pull request, so please ignore that dependency as of now.

**Comment by shubhrapahwa** [comment]
Hi,
I have updated the pull request.
Our jar is hosted here - https://oss.sonatype.org/#nexus-search;quick~dagcoin

But when I try to build, it doesn't look for this jar in the nexus repository, hence build fails.

Here is the screenshot of the same:
image

Please suggest how to configure gradle to download this jar while building.

Thanks!!

**Pull request closed without merging**